### PR TITLE
[ONNX] Remove unnecessary squeeze/unsqueeze when exporting batchnorm1d

### PR DESCRIPTION
**Pull request by daquexian**
ONNX BatchNorm op has supported N-d array (instead of only 4d array) since opset 6 (https://github.com/onnx/onnx/blob/master/docs/Changelog.md#BatchNormalization-6)

cc @BowenBao @neginraoof

### CODE REVIEW

**Comment by BowenBao** [comment]
looks like it's failing caffe2 test https://github.com/pytorch/pytorch/blob/bd88fd079310204e409e4995c351842a9bf3ec1c/test/onnx/test_pytorch_onnx_caffe2.py#L573

in

https://github.com/pytorch/pytorch/blob/bd88fd079310204e409e4995c351842a9bf3ec1c/caffe2/operators/spatial_batch_norm_op.h#L55

**Comment by spandantiwari** [comment]
Could you please address the failing Caffe2 tests. Thanks.

**Comment by garymm** [comment]
@daquexian is this still an issue?
I think it may have been fixed by #29458.

**Comment by daquexian** [comment]
> @daquexian is this still an issue?
> I think it may have been fixed by #29458.

I think you are right :)
